File: class07/cpp/AP0710-DepuracaoDivisao.cpp
#include <iostream>
#include <iomanip>
int main() {
    double a, b;
    std::cout << "Numerador: "; std::cin >> a;
    std::cout << "Denominador: "; std::cin >> b;
    if (b == 0) std::cout << "Erro: divisão por zero!" << std::endl;
    else std::cout << "Resultado: " << std::fixed << std::setprecision(2) << a/b << std::endl;
    return 0;
}

File: class07/python/AP0710-DepuracaoDivisao.py
# DepuracaoDivisao — Aula 07
# Programa com bug proposital: divisão por zero

a = float(input("Numerador: "))
b = float(input("Denominador: "))
# BUG CORRIGIDO: verificação antes da divisão
if b == 0:
    print("Erro: divisão por zero!")
else:
    print(f"Resultado: {a / b:.2f}")
